### PR TITLE
feat: reporter uses sources-server

### PR DESCRIPTION
**Pull request by andrii-codefresh**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: event_reporter/utils/app_version.go [hunk 1]
```diff
@@ -16,3 +19,17 @@ func RepoAppVersionsToEvent(applicationVersions *apiclient.ApplicationVersions)
 	}
 	return applicationVersionsEvents, nil
 }
+
+func SourcesAppVersionsToRepo(applicationVersions *sources_server_client.AppVersionResult, logCtx *log.Entry) *apiclient.ApplicationVersions {
+	if applicationVersions == nil {
+		return nil
+	}
+	applicationVersionsRepo := &apiclient.ApplicationVersions{}
+	applicationVersionsData, _ := json.Marshal(applicationVersions)
```
Comment (oleksandr-codefresh) [inline event_reporter/utils/app_version.go:28]: is this correct ? u encode and than decode same value

File: go.mod [hunk 2]
```diff
@@ -348,3 +348,6 @@ replace (
 	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.29.6
 	k8s.io/sample-controller => k8s.io/sample-controller v0.29.6
 )
+
+// TODO:CR-26144: Remove alias
+replace github.com/codefresh-io/octopus-argo => /Users/andrii/go/src/github.com/codefresh-io/octopus-argo
```
Comment (oleksandr-codefresh) [inline go.mod]: here is your local path

File: pkg/sources_server_client/client.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+type sourceServerClient struct {
+	clientConfig *SourcesServerConfig
+}
+
+type SourceServerClientInteface interface {
+	GetAppVersion(app *v1alpha1.Application, revisions *string) *AppVersionResult
+}
+
+func (c *sourceServerClient) sendRequest(method, url string, payload interface{}) ([]byte, error) {
```
Comment (oleksandr-codefresh) [inline pkg/sources_server_client/client.go:44]: method type missing

File: pkg/sources_server_client/client.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	var requestBody []byte
+	var err error
+	if payload != nil {
+		requestBody, err = json.Marshal(payload)
+		if err != nil {
+			return nil, fmt.Errorf("error marshalling payload: %w", err)
+		}
+	}
+
+	req, err := http.NewRequest(method, fmt.Sprintf("%s%s", c.clientConfig.BaseURL, url), bytes.NewBuffer(requestBody))
```
Comment (oleksandr-codefresh) [inline pkg/sources_server_client/client.go]: i guess we need somehow check if trailing slash present in passed URL

File: event_reporter/utils/app_version.go (hunk 1 above)
Comment (andrii-codefresh) [reply event_reporter/utils/app_version.go:28]: Yes. But to variable of another type

File: pkg/sources_server_client/client.go (hunk 3 above)
Comment (andrii-codefresh) [reply pkg/sources_server_client/client.go:44]: It is a `string` for both `method` and `url`

**Review by oleksandr-codefresh — APPROVED** [review]

**Pull request merged**